### PR TITLE
Implement quickSort() function

### LINKED ISSUE #11: Parameter --qsort
This one is may be not available to the standard sort of your OS.

### PR DESCRIPTION
**Pull request by yhabteab**
resolves #11

### CODE REVIEW

File: sort.php [hunk 1]
```diff
@@ -23,6 +23,50 @@ function checkForEquality($firstNum, $secondNum)
...
+    }
+
+    if ($arr[$i] > $pivot) {
+        swapElement($arr[$i], $arr[$highest]);
+    }
+
+    return $i;
+}
+
+function quickSort(&$arr, $lowest, $highest)
```
Comment (Al2Klimov) [inline sort.php]: Why do you pass $arr by reference here?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
`php sort.php --qsort -n in.txt` doesn't work as expected.

File: sort.php (hunk 1 above)
Comment (yhabteab) [reply sort.php]: I tried it out beforehand without returning the array and when it worked I forgot to put ``&`` it away.

**Comment by yhabteab** [comment]
And I didn't know that we also support quicksort ``--numeric-sort``. I will change it.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [ ] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [ ] `git commit --amend --no-edit`
* [ ] `git push -f origin feature/sort-#11`

**Comment by yhabteab** [comment]
##Testprotocol #11 

Since we have integrated all parameters for our bubble sort func, it is now time to implement ``quickSort()`` and to integrate the corresponding parameters.

The content of the in.txt file looks like this.
```
13
8
12
6
2
10
1
7
0
```
Commands with the ``--qsort`` and ``--numeric-sort`` parameter.
``php sort.php --qsort -n in.txt`` AND ``php sort.php --qsort -n < in.txt``  output:
```
0                                             0
1                                             1
2                                             2
6                                             6
7                                             7
8                                             8
10                                            10
12                                            12
13                                            13
```
and without ``--numeric-sort`` parameter the output looks like this..
``php sort.php --qsort in.txt`` AND ``php sort.php --qsort < in.txt``
```
0                                       0
1                                       1
10                                      10
12                                      12
13                                      13
2                                       2
6                                       6
7                                       7
8                                       8
```

See you next time!

Yonas :+1:

File: sort.php [hunk 2]
```diff
@@ -23,6 +23,31 @@ function checkForEquality($firstNum, $secondNum)
     return $firstNum < $secondNum;
 }
 
+function quickSort($arr, $funcCallback)
+{
+    $arrSize = sizeof($arr);
+
+    if ($funcCallback($arrSize, 2)) {
```
Comment (Al2Klimov) [inline sort.php]: Why $funcCallback?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: sort.php (hunk 2 above)
Comment (Yonas-net) [reply sort.php:30]: that's the name of the function that will give as callback

Comment (Al2Klimov) [reply sort.php]: Why are you using it in this case?

Comment (yhabteab) [reply sort.php]: because we call the function as a callback or maybe not?

Comment (Al2Klimov) [reply sort.php]: Why do you call this function in this case?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [ ] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [ ] `git commit --amend --no-edit`
* [ ] `git push -f origin feature/sort-#11`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol.

**Comment by yhabteab** [comment]
##Testprotocol #11 

Since we have integrated all parameters for our bubble sort func, it is now time to implement ``quickSort()`` and to 
integrate the corresponding parameters

The content of the in.txt file looks like this.
```
13
8
12
6
2
10
1
7
0
```
Commands with the ``--qsort`` and ``--numeric-sort`` parameter.
``php sort.php --qsort -n in.txt`` AND ``php sort.php --qsort -n < in.txt``  output:
```
0                                             0
1                                             1
2                                             2
6                                             6
7                                             7
8                                             8
10                                            10
12                                            12
13                                            13
```
and without ``--numeric-sort`` parameter the output looks like this..
``php sort.php --qsort in.txt`` AND ``php sort.php --qsort < in.txt``
```
0                                       0
1                                       1
10                                      10
12                                      12
13                                      13
2                                       2
6                                       6
7                                       7
8                                       8
```

See you next time!

:+1:

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**